src/modules/files-storage/api/uc/files-storage.uc.ts

```diff
@@ -188,6 +193,12 @@ export class FilesStorageUC {
 			responseType: 'stream',
 		};
 
+		const domain = new URL(params.url).hostname;
+		if (domain === this.config.FILES_STORAGE_OWN_DOMAIN) {
```

Comment:
why is this if necessary?
this looks quite hacky

also if you really need such an env var, consider changing that name

OWN_DOMAIN ... clarify what you do want here, either FILES_STORAGE_PUBLIC_URL or what?
And if you really need the domain just use domain, words like OWN and public in combination with domain is confusing

isn't there no better way to extract the jwt?